### PR TITLE
Planner mods

### PR DESCRIPTION
**Pull request by bminortx**
Changed the code in SimBA to reflect the relocation of the Protobufs and URDFParser to the HAL library.

### CODE REVIEW

File: Applications/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1,8 @@
-add_subdirectory( LocalSim )
-add_subdirectory( StateKeeper )
-add_subdirectory( Examples )
+add_subdirectory(LocalSim)
```
Comment (chachi) [inline Applications/CMakeLists.txt:1]: Do you want to just delete these lines to keep it clean?

File: Applications/LocalSim/LocalSim.h [hunk 2]
```diff
@@ -44,7 +41,7 @@ class LocalSim{
   SimDevices                  sim_devices_;
   RobotsManager               robot_manager_;
   NetworkManager              network_manager_;
-  URDF_Parser*                parser_;
+  URDFParser*                 parser_;
```
Comment (chachi) [inline Applications/LocalSim/LocalSim.h:45]: Would it help to make this a `shared_ptr` or `unique_ptr`?

File: Applications/MatlabInterface/GenMesh.m [hunk 3]
```diff
@@ -20,7 +20,7 @@
 z20 = bi2de(bitget(tau, 13:14), 'left-msb');
 z21 = bi2de(bitget(tau, 15:16), 'left-msb');
 z22 = bi2de(bitget(tau, 17:18), 'left-msb');
-Z = real([z00 z01 z02; z10 z11 z12; z20 z21 z22]);
+Z = real([z00 z01 z02; z10 z11 z12; z20 z21 z22])/4;
```
Comment (chachi) [inline Applications/MatlabInterface/GenMesh.m:23]: Might want a comment about why this is divided by 4

File: Applications/PathPlanner/PathPlanner.cpp [hunk 4]
```diff
@@ -92,6 +92,7 @@ void PathPlanner::SetHeightmap(pb::RegisterPlannerReqMsg& mRequest,
   GroundStates();
   mReply.set_success(1);
   mesh_set_ = true;
+  LOG(INFO) << "heightmap set!";
```
Comment (chachi) [inline Applications/PathPlanner/PathPlanner.cpp]: Do you want this/the rest of the LOG(INFO) statements still in here?

File: Applications/PathPlanner/PathPlanner.h [hunk 5]
```diff
@@ -114,11 +116,11 @@ class PathPlanner {
   // Our bitstring that describes the map.
   int  map_tau_;
   // booleans
-  bool config_set_;
-  bool mesh_set_;
-  bool policy_set_;
-  bool policy_delivered_;
-  bool policy_failed_;
+  volatile std::atomic<bool> config_set_;
```
Comment (chachi) [inline Applications/PathPlanner/PathPlanner.h]: If they're `std::atomic` (which I love), they don't need to be volatile. `std::atomic` takes care of that part for you.

File: SimBA/CMakeLists.txt [hunk 6]
```diff
@@ -67,9 +69,9 @@ list(APPEND MODELGRAPH_HDRS
   ModelGraph/GLDebugDrawer.h
   ModelGraph/GLHeightmap.h
   ModelGraph/GlutStuff.h
-  ModelGraph/ModelNode.h
+  # ModelGraph/ModelNode.h
```
Comment (chachi) [inline SimBA/CMakeLists.txt]: Does this still exist?

File: SimBA/CMakeLists.txt [hunk 7]
```diff
@@ -86,40 +88,40 @@ list(APPEND NETWORK_HDRS
...
-  SimRobots/SimRobot.h
-  SimRobots/SimWorld.h
-)
-
-list(APPEND URDFPARSER_HDRS
-  URDFParser/URDF_Parser.h
-  URDFParser/TinyXMLTool.h
-  URDFParser/GenURIFromURDF.h
-)
+# list(APPEND SIMDEVICES_HDRS
```
Comment (chachi) [inline SimBA/CMakeLists.txt]: Do all these still exist?

File: SimBA/CMakeLists.txt [hunk 8]
```diff
@@ -152,21 +154,22 @@ list(APPEND MODELGRAPH_SRCS
...
+# list(APPEND SIMDEVICES_SRCS
+#   SimDevices/SimDevices.cpp
+# )
 list(APPEND SIMDEVICES_CONTROL_SRCS)
 list(APPEND SIMDEVICES_SENSOR_SRCS)
 list(APPEND SIMROBOTS_SRCS)
-list(APPEND URDFPARSER_SRCS
-  URDFParser/URDF_Parser.cpp
-)
+# list(APPEND URDFPARSER_SRCS
```
Comment (chachi) [inline SimBA/CMakeLists.txt]: I think these are all gone now, right?

File: SimBA/Managers/NetworkManager.cpp [hunk 9]
```diff
@@ -648,7 +648,7 @@ bool NetworkManager::RegisterWithStateKeeper()
       XMLDocument doc;
       doc.Parse(urdf.xml().c_str());
       // create previous robot
-      URDF_Parser* parse = new URDF_Parser(debug_level_);
+      URDFParser* parse = new URDFParser(debug_level_);
```
Comment (chachi) [inline SimBA/Managers/NetworkManager.cpp]: Definitely a memory leak. Pointer needs to be deleted. In this case though, just make a stack object w/o `new`

File: SimBA/Managers/NetworkManager.cpp [hunk 10]
```diff
@@ -685,7 +685,7 @@ void NetworkManager::AddRobotByURDF(LocalSimAddNewRobotReqMsg& mRequest,
       mRequest.mutable_init_pose()->q(), mRequest.mutable_init_pose()->r();
 
   // add new robot in proxy
-  URDF_Parser* parse = new URDF_Parser(debug_level_);
```
Comment (chachi) [inline SimBA/Managers/NetworkManager.cpp:688]: Also a memory leak

File: SimBA/ModelGraph/EigenToBullet.h [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+#include <URDFParser/EigenHelpers.h>
+
+//////////////////////////////////////////////////////////
+///
+/// EIGEN-TO-BULLET-TO-EIGEN CONVERTERS
+///
+//////////////////////////////////////////////////////////
+
+inline Eigen::Matrix4d
+getInverseTransformation (const Eigen::Matrix4d &transformation)
```
Comment (chachi) [inline SimBA/ModelGraph/EigenToBullet.h]: No space before `(`

Comment (chachi) [inline SimBA/ModelGraph/EigenToBullet.h]: ALso, `eigen::Matrix4d::inverse()` is a builtin for this.

File: SimBA/ModelGraph/EigenToBullet.h [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+{
+  btTransform bT;
+  Eigen::Matrix<btScalar,4,4> eT = T.cast<btScalar>();
+  bT.setFromOpenGLMatrix(eT.data());
+  return bT;
+}
+
+inline btVector3 toBulletVec3(const Eigen::Vector3d& v)
+{
+  btVector3 bv;
```
Comment (chachi) [inline SimBA/ModelGraph/EigenToBullet.h:65]: I believe this could simplify to `return btVector3(v(0), v(1), v(2))`

**Comment by chachi** [comment]
Wow that's a lot of deleted code. Done looking over these.

**Comment by chachi** [comment]
Where do we stand with this guy?

**Comment by bminortx** [comment]
This builds for me, but I haven't tested it; something's wrong with my Node. I keep getting `E/Node.cpp:1276 Caught SIGSEGV` when I run anything with node in it. I think that's just my machine, though, so I'd say this is good to merge (unless anyone has other changes).

**Comment by bminortx** [comment]
Everything runs and builds now. I am an idiot. @sinaaghli or @chachi can you take a look really quick?

**Comment by bminortx** [comment]
bump

**Pull request closed without merging**